core/src/main/java/hudson/Util.java

```diff
@@ -1433,19 +1341,18 @@ private static boolean createSymlinkJava7(@Nonnull File baseDir, @Nonnull String
                     throw fileAlreadyExistsException;
                 }
             }
-            return true;
         } catch (UnsupportedOperationException e) {
-                return true; // no symlinks on this platform
-        } catch (FileSystemException e) {
-            if (Functions.isWindows()) {
+            PrintStream log = listener.getLogger();
+            log.print("Symbolic links are not supported on this platform");
+            Functions.printStackTrace(e, log);
+        } catch (IOException e) {
+            if (Functions.isWindows() && e instanceof FileSystemException) {
                 warnWindowsSymlink();
```

Comment:
The code used to suppress any `FileSystemExceptions` from being logged, but since most NIO exceptions ware instances of `FileSystemException` it seems like we want to at least log them. Should we suppress them on Windows after the first warning?

Reply:
Not sure. Try to follow the logging/throwing behavior of the original method as much as possible, I guess. The design was weird; would prefer `TaskListener` to go away and the method to throw an exception if it fails to create the symlink for any reason (let the caller decide what to do next).

Anyway, if we get rid of build number symlinks, there should no common callers of this method.

Reply:
> Anyway, if we get rid of build number symlinks, there should no common callers of this method.

You are referring to inside core right?  outside of core the maven project type uses symlinks to link from build/x/  to individual module builds - because it has a questionable design :-/